product_code/report_service/impl/src/main/scala/de/upb/cs/uc4/report/impl/ReportServiceImpl.scala

```diff
@@ -56,6 +65,23 @@ class ReportServiceImpl(
 
   lazy val validationTimeout: FiniteDuration = config.getInt("uc4.timeouts.validation").milliseconds
 
+  private lazy val absoluteCertificateEnrollmentHtml = new File(config.getString("uc4.pdf.certificateEnrollmentHtml"))
+  lazy val certificateEnrollmentHtml: String = if (absoluteCertificateEnrollmentHtml.exists()) {
+    Using(Source.fromFile(absoluteCertificateEnrollmentHtml)) { source =>
+      source.getLines().mkString("\n")
+    }.getOrElse("")
+  }
+  else {
+    Source.fromResource("certificateEnrollment.html").getLines().mkString("\n")
```

Comment:
Maybe Try-catch for IO-exceptions?

Reply:
There are two things we could do in a catch: Log and Write a hard coded string into it,  to prevent a failure of a service.
Good idea.

Reply:
No, you can catch right here.
If we would extend another call which needs this variable, we need to handle there exact the same problem.
Except we want o handle the problem there in a different way, but I would say there is no reason to.
If this fails, there is a major bug in the system, because this is the fallback case.

Reply:
Added scalas try catch and logging, in case of an exception the empty string `""` is returned. Feel free to exchange with another fallback string.

Reply:
The commit is now listed